libbeat/outputs/mode/mode_test.go

```diff
@@ -26,6 +26,18 @@ func enableLogging(selectors []string) {
 	}
 }
 
+// wait for f to execute. If the function f does not end
+// within one second the method returns the error.
+func wait(f func(), err error) error {
+	ch := make(chan error, 1)
+	go func() {
+		f()
+		ch <- nil
+	}()
+	time.AfterFunc(time.Second, func() { ch <- err })
+	return <-ch
+}
```

Comment:
We need this? `go test` already has timeouts. Default is rather large, but in case 'close' generates a deadlock the test will fail. Timeouts in tests are good source for flaky tests in not so powerful CI setups.